xwayland/xwm.c

```diff
@@ -1792,6 +1824,24 @@ struct wlr_xwm *xwm_create(struct wlr_xwayland *xwayland, int wm_fd) {
 	return xwm;
 }
 
+void wlr_xwayland_surface_set_minimized(struct wlr_xwayland_surface *surface,
+		bool minimized) {
+	const bool change = surface->minimized != minimized;
+	surface->minimized = minimized;
+
+	if (minimized) {
+		xsurface_set_wm_state(surface, ICCCM_ICONIC_STATE);
+	} else {
+		xsurface_set_wm_state(surface, ICCCM_NORMAL_STATE);
+	}
+
+	xsurface_set_net_wm_state(surface);
+	xcb_flush(surface->xwm->xcb_conn);
+	if (change) {
+	    wlr_signal_emit_safe(&surface->events.request_minimize, surface);
```

Comment:
`wlr_xwayland_surface_set_minimized` should not emit a `request_minimize` event. `request_minimize` should only be used to signal that a client requests to be minimized.

Reply:
So where shoult the request_minimize event go?

Reply:
See https://github.com/swaywm/wlroots/pull/2264/files#r443474523.